dao.py

```diff
@@ -292,6 +292,29 @@ def get_tournament_by_id(self, id):
         '''id must be an ObjectId'''
         return M.Tournament.load(self.tournaments_col.find_one({'_id': id}), context='db')
 
+    def get_tournament_by_id_and_match_id(self, tournament_id, match_id):
+        tourney_m = M.Tournament.load(self.tournaments_col.find_one({'_id': tournament_id}, {'matches', 1}), context='db')
+        for match in tourney_m.matches:
+            if match_id == match.id:
+                return match
+
+    def set_match_exclusion_by_tournament_id_and_match_id(self, tournament_id, match_id, excluded):
+        # TODO ENHANCE THIS ALGORITHM TO ONLY UPDATE MATCH
+        match_updated = False
+        new_matches = []
+        tourney_m = \
+            M.Tournament.load(self.tournaments_col.find_one({'_id': tournament_id}), context='db')
+
+        for match in tourney_m.matches:
+            if int(match_id) == int(match.match_id):
```

Comment:
you might want to try/catch this in case they pass in something that is not an int for match_id (alternatively, make sure you cast to int before you call this function). even if you do nothing, the error should be caught in server.py, but it's best to handle errors explicitly.